app/controllers/course/users_controller.rb

```diff
@@ -0,0 +1,72 @@
+class Course::UsersController < Course::ModuleController
+  load_and_authorize_resource :course_user, through: :course, parent: false
+  before_action :authorize_show!, only: [:index]
+  before_action :authorize_edit!, except: [:index, :create]
+  before_action :ensure_unregistered_user, only: :create
+
+  def index # :nodoc:
+  end
+
+  def create # :nodoc:
+    @course_user.workflow_state = :approved if current_course_user && current_course_user.staff?
```

Comment:
why you manually assign the workflow_state here, should we just call <code>@course_user.approve!</code> ?